src/main/java/org/folio/oaipmh/processors/MarcWithHoldingsRequestHelper.java

```diff
@@ -219,12 +222,8 @@ private void processBatch(Request request, Context context, Promise<Response> oa
         final SourceStorageSourceRecordsClient srsClient = new SourceStorageSourceRecordsClient(request.getOkapiUrl(),
           request.getTenant(), request.getOkapiToken());
 
-        Future<Map<String, JsonObject>> srsResponse = Future.future();
-        if (CollectionUtils.isNotEmpty(instances)) {
-          srsResponse = requestSRSByIdentifiers(srsClient, instancesWithoutLast, deletedRecordSupport);
-        } else {
-          srsResponse.complete();
-        }
+        Future<Map<String, JsonObject>> srsResponse = requestSRSByIdentifiers(srsClient, instancesWithoutLast, deletedRecordSupport);
```

Comment:
probably don't need this `srsResponse` object at all, just call `onSuccess(...)` directly on the future returned by `requestSRSByIdentifiers(...)`

Reply:
Yes, you are right. Thank you for your note.